src/System.Text.Json/src/System/Text/Json/JsonDocument.CustomStack.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json
+{
+    public sealed partial class JsonDocument
+    {
+        private struct CustomStack : IDisposable
```

Comment:
Nit: RowStack?  StackRowStack?  Or something along those lines.  "CustomStack" isn't very descriptive for its purpose.

Reply:
Went with StackRowStack.  Still sort of weird, but at least it conveys it's specialized :)

Reply:
StackRowStack isn't quite descriptive and somewhat confusing.

How about something like `JsonSizeInfoRow` and `JsonSizeInfoRowStack`? If that isn't long enough, maybe `JsonElementSizeInfoRow`? 😛